src/main/kotlin/link/infra/packwiz/installer/UpdateManager.kt

```diff
@@ -97,6 +99,105 @@ class UpdateManager internal constructor(private val opts: Options, val ui: IUse
 			handleCancellation()
 		}
 
+		// MultiMC MC and loader version checker
+		val multimcManifestPath = Paths.get(opts.multimcFolder, "mmc-pack.json").toString();
+		val multimcManifestFile = File(multimcManifestPath)
+
+		if (multimcManifestFile.exists()) {
+			ui.submitProgress(InstallProgress("Loading MultiMC pack file..."))
+
+			val multimcManifest = try {
+				JsonParser.parseReader(multimcManifestFile.reader())
+			} catch (e: JsonIOException) {
+				ui.showErrorAndExit("Cannot read the MultiMC pack file", e)
+			} catch (e: JsonSyntaxException) {
+				ui.showErrorAndExit("Invalid MultiMC pack file", e)
+			}.asJsonObject
+
+			Log.info("Loaded MultiMC config")
+
+			// We only support format 1, if it gets updated in the future we'll have to handle that
+			// There's only version 1 for now tho, so that's good
+			if (multimcManifest["formatVersion"].asInt != 1) {
+				ui.showErrorAndExit("Invalid MultiMC format version")
+			}
+
+			var manifestModified = false
+			val modLoaders = hashMapOf("net.minecraftforge" to "forge", "net.fabricmc.fabric-loader" to "fabric")
+			val modLoadersClasses = modLoaders.entries.associate{(k,v)-> v to k}
+			var modLoaderFound = false
+			var modLoader: String? = null
+			var modLoaderOldVer: String? = null
+			var mcOldVer = "Unknown"
+			val components = multimcManifest["components"].asJsonArray
+			for (componentObj in components) {
+				val component = componentObj.asJsonObject
+
+				val version = component["version"].asString
+				when (component["uid"].asString) {
```

Comment:
Don't need a `when` for this, might as well turn it into an `if`

Reply:
Actually it doesn't even need to be special-cased, you might as well just make it part of `modLoaders` and allow updating multiple